### ISSUE TITLE
parseStringAsEnum completely ignores default values set in the enum

### ISSUE THREAD

**Issue by tnichols217** [Nigecat/obsidian-desmos#67]
**Describe the bug**
When using the parseStringAsEnum function, when resolving RelativeCSSUnit.percent, it only recognises the string "percent" instead of the actual value of the enum "%"

**Expected behavior**
parseStringAsEnum parses "%" as RelativeCSSUnit.percent

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Obsidian Version: v14.0.6
 - Plugin Version: 0.6.3

**Additional context**
I didnt want to put this on the #64 thread because it is somewhat unrelated, but still a bug

**Comment by Nigecat** [comment]
Yeah the rest of the parsers match the key then use the enum value since a lot of the time I need a different value internally to the public value and since TS enum are basically just objects and when you reference a field you're just getting the value it doesn't really work very well other way in most cases.

Regardless though I see how you'll need it for #64 (and it'll be useful in general) so I'll add a method to parse the other way.

**Comment by Nigecat** [comment]
See https://github.com/Nigecat/obsidian-desmos/pull/68#issuecomment-1163837845
